BalancerRPi.ino

```diff
@@ -136,31 +136,44 @@ void driveAround()
 
 void standUp()
 {
+  int standupvalue = 1;
   motors.setSpeeds(0, 0);
   buzzer.play("!>grms>a16>a16>g2");
   ledGreen(1);
   ledRed(1);
   ledYellow(1);
   int distanceDiff = 0;
   while (buzzer.isPlaying());
-  for(int speed = 0; speed < MOTOR_SPEED_LIMIT - 15; speed++){
-    integrateEncoders();
-    distanceDiff = distanceLeft - distanceRight;    
-    motors.setSpeeds(-speed - 1 * distanceDiff, -speed + 1 * distanceDiff);
-    delay(3);
-  }
-  for(int i = 0; i < 50; i++) {
-    integrateEncoders();
-    distanceDiff = distanceLeft - distanceRight;    
-    motors.setSpeeds(-MOTOR_SPEED_LIMIT + 15 - 10 * distanceDiff, -MOTOR_SPEED_LIMIT + 15 + 10 * distanceDiff);
-    delay(1);
+
+  if (angle > 70000) {
+    standupvalue = 1;
+  } else if (angle < -70000) {
+    standupvalue = -1;
   }
-  motors.setSpeeds(MOTOR_SPEED_LIMIT, MOTOR_SPEED_LIMIT);
+  
+  for(int speed = 0; speed < MOTOR_SPEED_LIMIT - 15; speed++){
+      integrateEncoders();
+      distanceDiff = distanceLeft - distanceRight;    
+      motors.setSpeeds(standupvalue * -speed - 1 * distanceDiff, standupvalue * -speed + 1 * distanceDiff);
+      delay(3);
+    }
+    
+    for(int i = 0; i < 50; i++) {
+      integrateEncoders();
+      distanceDiff = distanceLeft - distanceRight;    
+      motors.setSpeeds(standupvalue * -MOTOR_SPEED_LIMIT + 15 - 10 * distanceDiff, standupvalue * -MOTOR_SPEED_LIMIT + 15 + 10 * distanceDiff);
```

Comment:
Here -MOTOR_SPEED_LIMIT + 15 is to give a margin for position correction based on distanceDiff. With your changes, when standupvalue is -1, you will get a speed > MOTOR_SPEED_LIMIT as a base